### PR TITLE
Return node builder initialization errors instead of throwing fatal errors

### PR DESCRIPTION
**Pull request by peterargue**
This is a first step towards improving unrecoverable error handling in nodes. In the current implementation, initialization errors are either handled by panicking or calling `log.Fatal()`, the latter resulting in a call to `os.Exit`.

To improve on this, this change refactors the node builder `Initialize()` and methods it calls to return errors back to the caller instead of triggering a fatal error/panic. This will allow the caller to override the default behavior by restarting the node or taking some other action.

### CODE REVIEW

File: cmd/access/node_builder/staked_access_node_builder.go [hunk 1]
```diff
@@ -67,7 +67,7 @@ func (fnb *StakedAccessNodeBuilder) InitIDProviders() {
 	})
 }
 
-func (builder *StakedAccessNodeBuilder) Initialize() cmd.NodeBuilder {
+func (builder *StakedAccessNodeBuilder) Initialize() error {
```
Comment (huitseeker) [inline cmd/access/node_builder/staked_access_node_builder.go:70]: Yup, the added bonus being that this makes the side-effecting behavior of the initialization explicit.

**Review by huitseeker — COMMENTED** [review]
This is a really nice, straightforward cleanup, thanks @peterargue !

File: cmd/collection/main.go [hunk 2]
```diff
@@ -90,62 +90,68 @@ func main() {
...
+		// epoch qc contract flags
+		flags.StringVar(&accessAddress, "access-address", "", "the address of an access node")
+		flags.StringVar(&secureAccessNodeID, "secure-access-node-id", "", "the node ID of the secure access GRPC server")
+		flags.BoolVar(&insecureAccessAPI, "insecure-access-api", true, "required if insecure GRPC connection should be used")
+	})
+
+	err = nodeBuilder.Initialize()
+	if err != nil {
+		nodeBuilder.Logger.Fatal().Err(err).Send()
+	}
```
Comment (smnzhu) [inline cmd/collection/main.go:152]: nit:

```suggestion
	if err = nodeBuilder.Initialize(); err != nil {
		nodeBuilder.Logger.Fatal().Err(err).Send()
	}
```

Simlilar suggestion for all the other places

File: cmd/scaffold.go [hunk 3]
```diff
@@ -790,7 +790,10 @@ func (fnb *FlowNodeBuilder) Initialize() NodeBuilder {
 
 	fnb.ParseAndPrintFlags()
 
-	fnb.extraFlagsValidation()
+	err := fnb.extraFlagsValidation()
+	if err != nil {
```
Comment (smnzhu) [inline cmd/scaffold.go:794]: same here

File: cmd/scaffold.go [hunk 4]
```diff
@@ -799,12 +802,15 @@ func (fnb *FlowNodeBuilder) Initialize() NodeBuilder {
 
 	if fnb.metricsEnabled {
 		fnb.EnqueueMetricsServerInit()
-		fnb.RegisterBadgerMetrics()
+		err = fnb.RegisterBadgerMetrics()
+		if err != nil {
```
Comment (smnzhu) [inline cmd/scaffold.go:806]: and here

**Review by smnzhu — APPROVED** [review]
This is awesome :)

**Review by huitseeker — APPROVED** [review]
Oops, I meant to approve that.

**Comment by peterargue** [comment]
bors merge

**Pull request merged**